msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -166,21 +163,39 @@ public Integer getNumberOfDataPoints() {
 
   @Override
   public MsSpectrumType getSpectrumType() {
-    MsSpectrumType spectrumType = SpectrumTypeDetectionAlgorithm.detectSpectrumType(mzValues,
-        intensityValues, getMzBinaryDataInfo().getArrayLength());
-    return spectrumType;
+    if (getCVValue(MzMLCV.cvCentroidSpectrum) != null)
+      return MsSpectrumType.CENTROIDED;
+
+    if (getCVValue(MzMLCV.cvProfileSpectrum) != null)
+      return MsSpectrumType.PROFILE;
+
+    return null;
   }
 
   @Override
   public Float getTIC() {
-    Float tic = MsSpectrumUtil.getTIC(intensityValues, getMzBinaryDataInfo().getArrayLength());
+    Float tic = null;
+    try {
+      tic = Float.valueOf(getCVValue(MzMLCV.cvTIC));
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    if (tic == null)
+      tic = MsSpectrumUtil.getTIC(getIntensityValues(), getMzBinaryDataInfo().getArrayLength());
     return tic;
   }
 
   @Override
   public Range<Double> getMzRange() {
-    Range<Double> mzRange =
-        MsSpectrumUtil.getMzRange(mzValues, getMzBinaryDataInfo().getArrayLength());
+    Range<Double> mzRange = null;
```

Comment:
the variables like mzRange and TIC have to be declared as global variables, otherwise they are set to null with each call to this method